docs/snippets/angular/component-story-custom-args-complex.ts.mdx

```diff
@@ -0,0 +1,46 @@
+```ts
+// YourComponent.stories.ts
+
+import { Story, Meta } from '@storybook/angular/types-6-0';
+import YourComponent from './your.component';
+
+// a function to apply some computations
+const someFunction = (valuePropertyA: string, valuePropertyB: string) => {
+  // makes some computations and returns something
+};
+
+export default {
+  component: YourComponent,
+  title: 'A complex case with a function',
+  // creates specific argTypes with options
+  argTypes: {
+    propertyA: {
+      control: {
+        type: 'select',
+        options: ['Item One', 'Item Two', 'Item Three']
+      },
+    },
+    propertyB: {
+      control: {
+        type: 'select',
+        options: ['Another Item One', 'Another Item Two', 'Another Item Three']
+      },
+    },
+  },
+} as Meta;
+
+const Template: Story<YourComponent> = (args: YourComponent) => {
+  const { propertyA, propertyB } = args;
+
+  const someFunctionResult = someFunction(propertyA, propertyB);
+
+  args.someProperty = someFunctionResult;
+  return {
+    component: YourComponent,
+    props: {
+      ...args,
+      someProperty:someFunctionResult
```

Comment:
```suggestion
      someProperty: someFunctionResult,
```

Reply:
Fixed with the latest commit. Thank you for checking it :)